engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/FilteredTableDataService.java

```diff
@@ -136,13 +136,22 @@ public boolean hasTableLocationKey(@NotNull final TableLocationKey tableLocation
 
         @Nullable
         @Override
-        public TableLocation getTableLocationIfPresent(@NotNull TableLocationKey tableLocationKey) {
+        public TableLocation getTableLocationIfPresent(@NotNull final TableLocationKey tableLocationKey) {
             if (!locationKeyFilter.accept(tableLocationKey)) {
                 return null;
             }
             return inputProvider.getTableLocationIfPresent(tableLocationKey);
         }
 
+        @Override
+        public void removeTableLocationKey(@NotNull final TableLocationKey locationKey) {
+            if (!locationKeyFilter.accept(locationKey)) {
+                return;
+            }
+
```

Comment:
Not obvious we should filter instead of removing it unconditionally.  Enterprise does not check before removing.